src/Avalonia.Themes.Fluent/Controls/SplitButton.xaml

```diff
@@ -0,0 +1,392 @@
+    <Setter Property="BorderBrush" Value="{DynamicResource SplitButtonBorderBrushChecked}" />
+  </Style>
+  <Style Selector="SplitButton:checked-secondary-pointerover /template/ Rectangle#PrimaryBackgroundRectangle">
+    <Setter Property="Fill" Value="{DynamicResource SplitButtonBackgroundChecked}" />
+  </Style>
+  <Style Selector="SplitButton:checked-secondary-pointerover /template/ Rectangle#SecondaryBackgroundRectangle">
+    <Setter Property="Fill" Value="{DynamicResource SplitButtonBackgroundCheckedPointerOver}" />
+  </Style>
+  <Style Selector="SplitButton:checked-secondary-pointerover /template/ Button#PART_PrimaryButton">
+    <Setter Property="Foreground" Value="{DynamicResource SplitButtonForegroundChecked}" />
+  </Style>
+  <Style Selector="SplitButton:checked-secondary-pointerover /template/ Button#PART_SecondaryButton">
+    <Setter Property="BorderBrush" Value="{DynamicResource SplitButtonBorderBrushCheckedPointerOver}" />
+    <Setter Property="Foreground" Value="{DynamicResource SplitButtonForegroundCheckedPointerOver}" />
+  </Style>
+  <Style Selector="SplitButton:checked-secondary-pointerover /template/ Path#DropDownGlyphPath">
+    <Setter Property="Fill" Value="{DynamicResource SplitButtonForegroundCheckedPointerOver}" />
+  </Style>
+
+  <!-- CheckedSecondaryPressed State -->
+  <Style Selector="SplitButton:checked-secondary-pressed /template/ Border#Border">
+    <Setter Property="BorderBrush" Value="{DynamicResource SplitButtonBorderBrushChecked}" />
+  </Style>
+  <Style Selector="SplitButton:checked-secondary-pressed /template/ Rectangle#PrimaryBackgroundRectangle">
+    <Setter Property="Fill" Value="{DynamicResource SplitButtonBackgroundChecked}" />
+  </Style>
+  <Style Selector="SplitButton:checked-secondary-pressed /template/ Rectangle#SecondaryBackgroundRectangle">
+    <Setter Property="Fill" Value="{DynamicResource SplitButtonBackgroundCheckedPressed}" />
+  </Style>
+  <Style Selector="SplitButton:checked-secondary-pressed /template/ Button#PART_PrimaryButton">
+    <Setter Property="Foreground" Value="{DynamicResource SplitButtonForegroundChecked}" />
+  </Style>
+  <Style Selector="SplitButton:checked-secondary-pressed /template/ Button#PART_SecondaryButton">
+    <Setter Property="BorderBrush" Value="{DynamicResource SplitButtonBorderBrushCheckedPressed}" />
+    <Setter Property="Foreground" Value="{DynamicResource SplitButtonForegroundCheckedPressed}" />
+  </Style>
+  <Style Selector="SplitButton:checked-secondary-pressed /template/ Path#DropDownGlyphPath">
+    <Setter Property="Fill" Value="{DynamicResource SplitButtonForegroundCheckedPressed}" />
+  </Style>
+
+  <!-- SecondaryButtonPlacement States -->
+  <!-- Note: ':secondary-button-right' is the default -->
+  <!--<Style Selector="SplitButton:secondary-button-span /template/ Button#PART_SecondaryButton">
+    <Setter Property="(Grid.Column)" Value="0" />
+    <Setter Property="(Grid.ColumnSpan)" Value="3" />
```

Comment:
It'll have been with the setting of `Grid.Column` or `Grid.ColumnSpan` and not the design of spanning the secondary button, but what specifically was not working I don't remember. I know the Grid attached properties always give me issues in styles

> Full disclosure: I'm not entirely sure how the secondary button spanning the full width is supposed to work. I realize that for touch devices the larger (full width) hit target is ideal. However, that would also seem to overlap the primary area and render the primary button useless (if not also visually occluded).

My guess is that the design team at MS decided that it's a better experience for touch users to make them select from the drop down rather than the shortcut top level button, so touch basically just turns this into a drop down button. Whether that's ideal or not is going is a subjective opinion.

Reply:
Thanks for the info. Your assessment of Microsoft's thinking is probably pretty close to accurate. I think for now I'll leave the functionality in code but disabled with comments based on the info here.

For Avalonia, it may not always make sense to disable the primary button like that on touch-first platforms. That thinking could change in the future though.

Reply:
Attached Grid properties aren't set using this syntax for unknown reasons. I updated the code though and then disabled it as discussed.

Reply:
Touch input changes to functionality were entirely removed in later commits. As it was, this was already disabled in code due to above mentioned issues. Marking resolved.

```csharp
            // In WinUI, the span of the secondary button is changed to full-width for touch-based
            // devices in certain conditions. The full reasoning for this is unknown. Some theories
            // include:
            //
            //   My guess is that the design team at MS decided that it's a better experience
            //   for touch users to make them select from the drop down rather than the shortcut
            //   top level button, so touch basically just turns this into a drop down button.
            //   Whether that's ideal or not is going is a subjective opinion.
            //
            // For Avalonia, it may not always make sense to disable the primary button like that
            // on touch-first platforms. Users and developers would normally expect a control to
            // function the same on all platforms. Therefore, this functionality is disabled here
            // but could be re-enabled in the future if more reasons become known.
```